src/runtime/interop.cs

```diff
@@ -522,4 +517,27 @@ public Thunk(Delegate d)
             fn = d;
         }
     }
+
+    internal struct ThunkInfo
+    {
+        public Delegate Target;
+        public IntPtr Address;
+
+        public static readonly ThunkInfo Empty = new ThunkInfo(null, IntPtr.Zero);
+
+        public ThunkInfo(Delegate target, IntPtr addr)
+        {
+            Target = target;
+            Address = addr;
+        }
+    }
+    
+    [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Ansi)]
```

Comment:
NIT: none of the fields have strings, so why does `CharSet` matter?

Reply:
It doesn't, I'll remove it.

Reply:
It doesn't, I'll remove it.